lib/heartbeat.ex

```diff
@@ -22,6 +26,7 @@ defmodule OffBroadwayRedisStream.Heartbeat do
 
   @impl true
   def handle_continue(nil, state) do
+    create_group(state)
```

Comment:
Since creating group is async now there can be an edge case due to race condition between producer and heartbeat right?

If group is not created yet,
`Heartbeat:init` -> `Producer:init` ->  `Producer:handle_deamnd` (crash) -> `Heartbeat:create_group`

It should be fine with supervisor, though.

I am not sure how difficult it is to write a test for this, but if possible, can you add a test for this?

Reply:
@akash-akya I added a test that can help simulate this situation. I also simulated it with a small test app by adding `Process.sleep(5000)` in Heartbeat's init. Lemme know what you think.

Reply:
This is perfect! thanks for checking